r/tests/testthat/test-dataset-csv.R

```diff
@@ -299,3 +299,20 @@ test_that("open_dataset() deals with BOMs (byte-order-marks) correctly", {
     tibble(a = c(1, 3), b = c(2, 4))
   )
 })
+
+test_that("Error if read_options$column_names and schema-names differ (ARROW-14744)", {
+
+  dst_dir  <- make_temp_dir()
+  dst_file <- file.path(dst_dir, "file.csv")
+  df <- df1[, c("int", "dbl")]
+  write.csv(df, dst_file, row.names = FALSE, quote = FALSE)
+
+  # Mismatch of column names vs. schema-names should raise an error
+  schema  <- schema(int = int32(), dbl = float64())
+
+  expect_error(
+    open_dataset(csv_dir, format = "csv", schema = schema, column_names = c("these", "wont", "match")),
+    "column_names must match schema-names"
```

Comment:
```suggestion
    "Values in `column_names` must match schema field names"
```
Updating this to match change made above